lib/spack/spack/build_systems/oneapi.py

```diff
@@ -47,6 +49,17 @@ def install(self, spec, prefix, installer_path=None):
             installer_path = basename(self.url_for_version(spec.version))
 
         if platform == 'linux':
+            # Intel installer assumes and enforces that all components
+            # are installed into a single prefix. It knows about other
+            # installs by saving info in /var/intel/installercache
+            # (for root) and $HOME/intel/installercache (non root). To
+            # keep this install from interfering with other installs
+            # in spack and external to spack, delete the
+            # installercache when running as root or redefining HOME
```

Comment:
I understand the "when running as root" part but not the "or redefining HOME" part: I don't see any checks for redefining `HOME` added in this PR.

Other than the confusion with the comment, this looks good to me (but could you clarify what you meant there?)

Reply:
The code for redefining the HOME env variable was already there: https://github.com/spack/spack/blob/d87b8c9d54a7651dda04ebb73c515a108ca59a4d/lib/spack/spack/build_systems/oneapi.py#L65

Some good news is that they are adding a command line switch to specify the location where the installer drops the files. We will still need to support the old behavior, but it can be conditional on the installer version.

Reply:
To clarify what confuses me, the line

> delete the installercache when running as root or redefining HOME

Implies to me that the installercache will be deleted if there is a redefinition of `HOME`. I don't see that happening here (I only see it being removed if the user is `root`).

Reply:
I rewrote the entire paragraph. Please let me know if it is better.

Reply:
That looks great to me - thanks!